dcos/jsonitem.py

```diff
@@ -64,10 +64,10 @@ def find_parser(key, schema):
 
     key_schema = schema['properties'].get(key)
     if key_schema is None:
-        keys = ', '.join(schema['properties'].keys())
+        keys = '\n'.join(schema['properties'].keys())
         raise DCOSException(
-            'Error: {!r} is not a valid property. '
-            'Possible properties are: {}'.format(key, keys))
+            'Property {!r} is invalid - '
+            'possible properties are: \n{}'.format(key, keys))
```

Comment:
We can't really factor it out since the other version outputs the keys as core.foo while this one outputs foo. Also the beginning part of the string is different so the formatting function would be fairly small.